File: 2022/05/16/rangeSumQuery.py
"""
https://leetcode.com/problems/range-sum-query-immutable/
Author: Andrew Jarombek
Date: 5/16/2022
"""

from typing import List


class NumArray:
    def __init__(self, nums: List[int]):
        self.nums = nums

    def sum_range(self, left: int, right: int) -> int:
        return sum(self.nums[left:right + 1])

File: 2023/02/07/go/range_sum_query.go
/**
 * https://leetcode.com/problems/range-sum-query-immutable/
 * Author: Andrew Jarombek
 * Date: 2/7/2023
 */

package _go

type NumArray struct {
	Values []int
}

func Constructor(nums []int) NumArray {
	return NumArray{Values: nums}
}

func (numArray *NumArray) SumRange(left int, right int) int {
	result := 0

	for i := left; i <= right; i++ {
		result += numArray.Values[i]
	}

	return result
}
